api/apps/api/src/modules/scenarios/input-parameter-file.provider.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { isDefined } from '@marxan/utils';
+import { ScenariosCrudService } from './scenarios-crud.service';
+
+class InputParameterFile {
+  constructor(
+    private readonly boundaryLengthModifier?: number,
+    private readonly numberOfRuns?: number,
+    private readonly marxanInputParameterFile?: Record<string, string | number>,
+  ) {}
+
+  toString(): string {
```

Comment:
if it shall serve the real `input.dat` file, it probably needs headers:

```
Input file for Annealing program.

This file generated by ArcMarxan Toolbox version 1.0.3
created by Apropos Information Systems Inc.

General Parameters
VERSION 0.1
BLM 1.00000000000000E+0000
PROP 5.00000000000000E-0001
RANDSEED -1
NUMREPS 100

Annealing Parameters
NUMITNS 1000000
STARTTEMP -1.00000000000000E+0000
COOLFAC -1.00000000000000E+0000
NUMTEMP 10000

Cost Threshold
COSTTHRESH 0.00000000000000E+0000
THRESHPEN1 0.00000000000000E+0000
THRESHPEN2 0.00000000000000E+0000

Input Files
INPUTDIR input
SPECNAME spec.dat
PUNAME pu.dat
PUVSPRNAME puvsp.dat
BOUNDNAME bound.dat
MATRIXSPORDERNAME puvsp_sporder.dat

Save Files
SCENNAME output
SAVERUN 3
SAVEBEST 3
SAVESUMMARY 3
SAVESCEN 3
SAVETARGMET 3
SAVESUMSOLN 3
SAVELOG 3
SAVESNAPSTEPS 0
SAVESNAPCHANGES 0
SAVESNAPFREQUENCY 0
OUTPUTDIR output

Program control.
RUNMODE 1
MISSLEVEL 9.50000000000000E-0001
ITIMPTYPE 1
HEURTYPE -1
CLUMPTYPE 0
VERBOSITY 2
SAVESOLUTIONSMATRIX 3
```

not sure tho, @aagm ?

Reply:
there's no need for anything else than the known `KEY value` pairs - the marxan cli iterates over the lines of `input.dat` and extracts the known `KEY value` pairs - everything else is ignored.